### PR TITLE
feat: adds support to laravel 7 & supports phpstan 12

### LINKED ISSUE #381: Update PHPStan to 0.12.* version
- Larastan Version: 0.4.3

### Description

Can you, please, update phpstan from 0.11.19 version to the next minor version ^0.12.*?

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request prepares Larastan for Laravel 7.

Also, allows to start preparing Collision for Laravel 7. https://github.com/nunomaduro/collision/pull/72

Closes #381

### CODE REVIEW

**Comment by szepeviktor** [comment]
For PHP 7.1 you could add `"orchestra/testbench": "^3.9.2 | dev-master",`

**Comment by nunomaduro** [comment]
Thanks @szepeviktor. Also added better testing against different laravel versions.

**Comment by nunomaduro** [comment]
I actually may turn this pull request into the next version of 0.5. That targets the next version of PHPStan.

File: composer.json [hunk 1]
```diff
@@ -13,20 +13,20 @@
...
+        "illuminate/support": "5.6.*|5.7.*|5.8.*|^6.0|^7.0",
         "mockery/mockery": "^1.0|0.9.*",
         "phpstan/phpstan": "^0.11.15",
-        "symfony/process": "^4.2",
+        "symfony/process": "^4.2|^5.0",
         "ext-json": "*"
     },
     "require-dev": {
-        "orchestra/testbench": "^3.8",
+        "orchestra/testbench": "^3.6|dev-master",
```
Comment (GrahamCampbell) [inline composer.json]: No version of testbench will work properly yet, until https://github.com/orchestral/testbench-core/pull/32 is merged.

**Comment by canvural** [comment]
> I actually may turn this pull request into the next version of 0.5. That targets the next version of PHPStan.

A lot of stuff changed in 0.12  I actually have an incomplete branch that upgrades to 0.12 I can also push that if you want.

File: composer.json [hunk 2]
```diff
@@ -13,20 +13,20 @@
...
         "php": "^7.1.3",
         "composer/composer": "^1.0",
-        "illuminate/console": "5.6.*|5.7.*|5.8.*|^6.0",
-        "illuminate/container": "5.6.*|5.7.*|5.8.*|^6.0",
-        "illuminate/contracts": "5.6.*|5.7.*|5.8.*|^6.0",
-        "illuminate/database": "5.6.*|5.7.*|5.8.*|^6.0",
-        "illuminate/http": "5.6.*|5.7.*|5.8.*|^6.0",
-        "illuminate/pipeline": "5.6.*|5.7.*|5.8.*|^6.0",
-        "illuminate/support": "5.6.*|5.7.*|5.8.*|^6.0",
+        "illuminate/console": "5.6.*|5.7.*|5.8.*|^6.0|^7.0",
```
Comment (canvural) [inline composer.json]: We can also drop support for 5.6 and 5.7 I think.

**Comment by nunomaduro** [comment]
@canvural can pull request your branch against this one?

File: composer.json (hunk 2 above)
Comment (mfn) [reply composer.json]: 😢 

I guess it only gets harder for project supporting older versions:
https://github.com/rebing/graphql-laravel/blob/223642f36e444b931ba62e57a032ffa851f9b4fe/composer.json#L37-L38

I wonder though if it's possible, similar to phpstan shim, to have something alike for larastan.

**Comment by canvural** [comment]
@nunomaduro Done. #378

**Comment by GrahamCampbell** [comment]
I'd say only support for Laravel 6.0 LTS and 7.0 will be relevant in 2020. People using 5.x can just use the older version of larastan.

**Comment by szepeviktor** [comment]
PHPStan 0.12 is a breakthrough!

**Comment by nunomaduro** [comment]
@szepeviktor @canvural I will be running some tests here now, to see if everything works as expected 👍

**Comment by nunomaduro** [comment]
@canvural Thinking about removing the artisan command, and merge this https://github.com/nunomaduro/larastan/pull/387. And targeting a major `0.5`. What do you think?

**Pull request merged**